src/Microsoft.AspNetCore.Mvc.Razor.Host/MvcRazorHost.cs

```diff
@@ -297,6 +315,33 @@ public override RazorParser DecorateRazorParser(RazorParser razorParser, string
             return new MvcRazorParser(razorParser, inheritedChunkTrees, DefaultInheritedChunks, ModelExpressionType);
         }
 
+        public override RazorParser DecorateRazorParser(
+            RazorParser incomingRazorParser,
+            TemplateEngineContext context)
```

Comment:
What benefit does having the extra surface area provide? Why is it cleaner than simply passing the context's relative path and reusing existing methods?